src/orion/core/cli/db/setup.py

```diff
@@ -53,11 +54,36 @@ def main(*args):
         if cancel.strip().lower() == "n":
             return
 
-    _type = ask_question("Enter the database type: ", "mongodb")
-    name = ask_question("Enter the database name: ", "test")
-    host = ask_question("Enter the database host: ", "localhost")
-
-    config = {"database": {"type": _type, "name": name, "host": host}}
+    # Get database type.
+    while True:
+        _type = (
+            ask_question(
+                "Enter the database type (available: {}): ".format(
+                    ", ".join(Database.typenames)
+                ),
+                "mongodb",
+            )
+            .strip()
+            .lower()
+        )
+        if _type in Database.typenames:
+            break
+        print(
+            "Unknown database type: {}, expected: {}".format(
+                _type, ", ".join(Database.typenames)
+            )
+        )
+        print()
+    # Get database arguments.
+    db_class = Database.types[Database.typenames.index(_type)]
+    db_args = db_class.get_arguments()
+    arg_vals = {}
+    for arg_name, default_value in db_args.items():
```

Comment:
I would sort the arguments here to make sure the order is deterministic.

```suggestion
    for arg_name, default_value in sorted(db_args.items()):
```

Reply:
Done !